### PR TITLE
Batching partition creation

### PR DESCRIPTION
**Pull request by vineethvarghese**

### CODE REVIEW

File: maestro-scalding/src/main/scala/au/com/cba/omnia/maestro/hive/HiveTable.scala [hunk 1]
```diff
@@ -60,13 +60,15 @@ sealed trait HiveTable[T <: ThriftStruct, ST] {
...
+  externalPath: Option[String] = None, partitionBatchSize: Int = 100
 ) extends HiveTable[A, (B, A)] {
 
   /** List of partition column names and type (string by default). */
-  val partitionMetadata: List[(String, String)] = partition.fieldNames.map(n => (n, "string"))
+  val partitionMetadata    = partition.fieldNames.map(n => (n, "string"))
   //Enforce the Hive partition pattern
-  val hivePartitionPattern: String = partition.fieldNames.map(_ + "=%s").mkString("/")
+  val hivePartitionPattern = partition.fieldNames.map(_ + "=%s").mkString("/")
+  val hdfsPartitionGlob    = partition.fieldNames.map(_=> "*").mkString("/")
```
Comment (rowandavies) [inline maestro-scalding/src/main/scala/au/com/cba/omnia/maestro/hive/HiveTable.scala]: space before `=>`

Comment (vineethvarghese) [inline maestro-scalding/src/main/scala/au/com/cba/omnia/maestro/hive/HiveTable.scala]: ok

File: maestro-scalding/src/main/scala/au/com/cba/omnia/maestro/hive/HiveTable.scala [hunk 2]
```diff
@@ -74,13 +76,21 @@ case class PartitionedHiveTable[A <: ThriftStruct : Manifest, B : Manifest : Tup
   override def writeExecution(pipe: TypedPipe[A], append: Boolean = true): Execution[ExecutionCounters] = {
     def modifyConfig(config: Config) = ConfHelper.createUniqueFilenames(config)
 
+    def addPartitions(paths: List[Path]): Execution[Unit] =
+      paths.grouped(partitionBatchSize)
+        .map((ps: List[Path]) => Execution.hive(Hive.addPartitions(database, table, partition.fieldNames, ps)))
+        .toList.sequence.map(_ => ())
```
Comment (rowandavies) [inline maestro-scalding/src/main/scala/au/com/cba/omnia/maestro/hive/HiveTable.scala]: You can use `sequence_` instead which avoids the need for the map, and doesn't create the list either.

Comment (rowandavies) [inline maestro-scalding/src/main/scala/au/com/cba/omnia/maestro/hive/HiveTable.scala]: See: http://docs.typelevel.org/api/scalaz/nightly/index.html#scalaz.Traverse

**Comment by rowandavies** [comment]
Looks good to me.  :+1:

File: maestro-scalding/src/main/scala/au/com/cba/omnia/maestro/hive/HiveTable.scala (hunk 2 above)
Comment (vineethvarghese) [inline maestro-scalding/src/main/scala/au/com/cba/omnia/maestro/hive/HiveTable.scala]: Done. Thanks

**Pull request merged**